Content.Client/Materials/UI/MaterialStorageControl.xaml.cs

```diff
@@ -0,0 +1,92 @@
+using System.Linq;
+using Content.Shared.Materials;
+using Robust.Client.AutoGenerated;
+using Robust.Client.UserInterface.Controls;
+using Robust.Client.UserInterface.XAML;
+using Robust.Shared.Timing;
+
+namespace Content.Client.Materials.UI;
+
+/// <summary>
+/// This widget is one row in the lathe eject menu.
+/// </summary>
+[GenerateTypedNameReferences]
+public sealed partial class MaterialStorageControl : BoxContainer
+{
+    [Dependency] private readonly IEntityManager _entityManager = default!;
+
+    private EntityUid? _owner;
+
+    private Dictionary<string, int> _currentMaterials = new();
+
+    public MaterialStorageControl()
+    {
+        RobustXamlLoader.Load(this);
+        IoCManager.InjectDependencies(this);
+    }
+
+    public void SetOwner(EntityUid owner)
+    {
+        _owner = owner;
+    }
+
+    protected override void FrameUpdate(FrameEventArgs args)
```

Comment:
so the one thing that's kinda sus about this code is that it doesn't use BUI states or events to update the control. It just runs in frame update and checks for potential changes to the amount of materials. I'm not sure if this is really the best way of doing this but i'm open to opinions.